cypress/utils/formFiller.ts

```diff
@@ -0,0 +1,272 @@
+};
+
+export const saveDraft = () => {
+  cy.get('[data-testid=save-draft-button]').click();
+};
+
+export const fillForm2 = (hankeData: HankeDataDraft) => {
+  if (hankeData.omistajat && hankeData.omistajat[0]) {
+    cy.get('input[data-testid=omistajat-etunimi]').type(hankeData.omistajat[0].etunimi);
+    cy.get('input[data-testid=omistajat-sukunimi]').type(hankeData.omistajat[0].sukunimi);
+    cy.get('input[data-testid=omistajat-email]').type(hankeData.omistajat[0].email);
+    cy.get('input[data-testid=omistajat-puhelinnumero]').type(hankeData.omistajat[0].puhelinnumero);
+  }
+};
+
+export const fillForm3 = (hankeData: HankeDataDraft) => {
+  if ('tyomaaKatuosoite' in hankeData && typeof hankeData.tyomaaKatuosoite === 'string') {
+    cy.get('input[data-testid=tyomaaKatuosoite]').type(hankeData.tyomaaKatuosoite);
+  }
+};
+
+export const selectKaistaHaitta = (kaistaHaitta: HANKE_KAISTAHAITTA_KEY) => {
+  cy.get('#kaistaHaitta-toggle-button').click();
+  switch (kaistaHaitta) {
+    case HANKE_KAISTAHAITTA.YKSI:
+      cy.get('#kaistaHaitta-item-0').click();
+      break;
+    case HANKE_KAISTAHAITTA.KAKSI:
+      cy.get('#kaistaHaitta-item-1').click();
+      break;
+    case HANKE_KAISTAHAITTA.KOLME:
+      cy.get('#kaistaHaitta-item-2').click();
+      break;
+    case HANKE_KAISTAHAITTA.NELJA:
+      cy.get('#kaistaHaitta-item-3').click();
+      break;
+    case HANKE_KAISTAHAITTA.VIISI:
+      cy.get('#kaistaHaitta-item-4').click();
+      break;
+    default:
+      break;
+  }
+};
+
+export const selectKaistanPituusHaitta = (kaistanPituusHaitta: HANKE_KAISTAPITUUSHAITTA_KEY) => {
+  cy.get('#kaistaPituusHaitta-toggle-button').click();
+  switch (kaistanPituusHaitta) {
+    case HANKE_KAISTAPITUUSHAITTA.YKSI:
+      cy.get('#kaistaPituusHaitta-item-0').click();
+      break;
+    case HANKE_KAISTAPITUUSHAITTA.KAKSI:
+      cy.get('#kaistaPituusHaitta-item-1').click();
+      break;
+    case HANKE_KAISTAPITUUSHAITTA.KOLME:
+      cy.get('#kaistaPituusHaitta-item-2').click();
+      break;
+    case HANKE_KAISTAPITUUSHAITTA.NELJA:
+      cy.get('#kaistaPituusHaitta-item-3').click();
+      break;
+    case HANKE_KAISTAPITUUSHAITTA.VIISI:
+      cy.get('#kaistaPituusHaitta-item-4').click();
+      break;
+    default:
+      break;
+  }
+};
+
+export const selectMeluHaitta = (meluHaitta: HANKE_MELUHAITTA_KEY) => {
+  cy.get('#meluHaitta-toggle-button').click();
+  switch (meluHaitta) {
+    case HANKE_MELUHAITTA.YKSI:
+      cy.get('#meluHaitta-item-0').click();
+      break;
+    case HANKE_MELUHAITTA.KAKSI:
+      cy.get('#meluHaitta-item-1').click();
+      break;
+    case HANKE_MELUHAITTA.KOLME:
+      cy.get('#meluHaitta-item-2').click();
+      break;
+    default:
+      break;
+  }
+};
+
+export const selectPolyHaitta = (polyHaitta: HANKE_POLYHAITTA_KEY) => {
+  cy.get('#polyHaitta-toggle-button').click();
+  switch (polyHaitta) {
+    case HANKE_POLYHAITTA.YKSI:
+      cy.get('#polyHaitta-item-0').click();
+      break;
```

Comment:
Hitsi olis kivempi tehdä geneerisempi testifunktio kaikille select-komponenteille ja niiden itemeille. 

Jotain tähän tyyliin, mutta toki vaatii muutoksen myös komponenttiin. Kannattanee ainakin pohtia ( :
```
select = (name, value) => {
  cy.get('#${name}-toggle-button').click();
  cy.get('#${name}-item-${value}').click();
}
```

Reply:
Hmm 🤔 
Totta, totta...
Pitäis mäpätä nuo mahdolliset avaimet "YKSI" | "KAKSI" | "KOLME".. noihin listan indekseihin.
Se logiikka pitäis lisätä johonkin -- ja toisaalta varmistautua siitä että jokaisen haitan avain tulisi pysymään samana, eikö näin?
Pohdin tätä, tuntui hyvältä ajatukselta alkuun, mutta aloin miettimään noiden hankekohtaisten avainten kääntämistä indekseihin ... Tjoo, en näe suorilta tässä vaiheessa että testikoodin yleistämistä olisi merkittävä hyöty. 🤔
Kommentti huomioitu, pohdittu, en vie eteenpäin. :)